test/cpp/tensorexpr/test_llvm.cpp

```diff
@@ -1400,34 +1399,31 @@ void DISABLED_testLLVMRFactorVectorizedReduction() {
 
   Placeholder a("a", kFloat, {1, M, N});
 
-  // TODO: why doesn't implicit vector<DimArg> work?
-  std::vector<DimArg> axis = {DimArg(1)};
-  std::vector<DimArg> reduce_axis = {DimArg(M), DimArg(N)};
-  Tensor* b = Reduce("sum", axis, Sum(), a, reduce_axis);
+  Tensor* b = Reduce("sum", {{1, "K"}}, Sum(), a, {{M, "M"}, {N, "N"}});
   LoopNest loopnest({b});
   std::vector<For*> loops = loopnest.getLoopStmtsFor(b);
   For* loop_k = loops.at(0);
   For* loop_m = loops.at(1);
   For* loop_n = loops.at(2);
-  loopnest.reorderAxis(loop_n, loop_m);
-  loops = loopnest.getLoopStmtsFor(b);
+  loopnest.rfactor(b->body(), loop_n->var());
+
+  loops = NodeFinder<For>::find(loopnest.root_stmt());
   loop_k = loops.at(0);
-  loop_n = loops.at(1);
+  // loop 1 is the initializer of tmp_buf
   loop_m = loops.at(2);
-  // Case-III reductions
-  loopnest.rfactor(b->body(), loop_n->var());
-  loopnest.prepareForCodegen();
-  Stmt* s = loopnest.root_stmt();
-  s = IRSimplifier::simplify(s);
+  loop_n = loops.at(3);
+  loopnest.reorderAxis(loop_n, loop_m);
 
-  Block* root_block = dynamic_cast<Block*>(s);
-  auto I = root_block->begin();
-  ++I;
+  // Case-III reductions
```

Comment:
Are the reductions cases publicly documented somewhere? If not, it's probably better to explain what the Case-III means.

Reply:
Not sure, this isn't my terminology - it was in the original disabled test.